src/plugins/elasticsearch/lib/call_with_request.js

```diff
@@ -10,7 +10,8 @@ module.exports = (client) => {
     }
     const api = _.get(client, endpoint);
     if (!api) throw new Error(`callWithRequest called with an invalid endpoint: ${endpoint}`);
-    return api.call(client, params)
+    const apiContext = endpoint.split('.')[0] === 'transport' ? client.transport : client;
```

Comment:
I'd feel a lot better about this if it used `var toPath = require('lodash/internal/toPath');` to parse the path and then did something like this:
```js
const path = toPath(endpoint);
const api = get(client, path);
const apiContext = get(client, path.slice(0, -1), client);
return api.call(apiContext, ...
```